src/lib.rs

```diff
@@ -101,12 +102,12 @@ impl<T: Copy + Hash + Eq> InvertableTransform for BlockId<T> {
 /// rather than vectors are provided.
 impl BlockId<char> {
     /// Encode a `u64` into an opaque string.
-    pub fn encode_string(&self, v: u64) -> String {
-        self.forward(v).into_iter().collect()
+    pub fn encode_string(&self, v: u64) -> Option<String> {
+        Some(self.forward(v)?.into_iter().collect())
     }
 
     /// Decode a `u64` from an opaque string.
-    pub fn decode_string(&self, v: &str) -> u64 {
+    pub fn decode_string(&self, v: &str) -> Option<u64> {
```

Comment:
minor: would a `Result` make more sense than an `Option`? If you pass an invalid string for decoding, I would think of the decoding attempt as having failed not "succeeded, but with a null output" which an option kinda implies? What do you think?

Reply:
I tend to prefer `Option` to `Result` when there is only one type of failure (e.g. indexing out of bounds in a vec), which is essentially the case here.